src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,271 @@
+    private Ui ui;
+
+    /**
+     * Constructor to initialise a Duke object.
+     *
+     * @param path File path to the text file which stores task information.
+     */
+    public Duke(String path) {
+        this.ui = new Ui();
+        this.storage = new Storage(path);
+
+        try {
+            tasks = new TaskList(storage.loadTaskList());
+        } catch (DukeException e) {
+            ui.printError(e);
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Starts the Duke chatbot to allow for user commands.
+     */
+    public void run() {
+        try {
+            ui.printGreeting();
+            storage.createDataFile();
+        } catch (Exception e) {
+            ui.printError(e);
+        }
+
+        while (hasNext) {
+            try {
+                String input = ui.getInput();
+                String command = Parser.parseCommand(input);
+                String info = "";
+
+                switch (command.toUpperCase()) {
+                case "BYE":
+                    handleExit();
+                    storage.update(tasks);
+                    break;
+                case "LIST":
+                    handleList();
+                    break;
+                case "MARK":
+                    info = Parser.parseInfo(input);
+                    handleMarking(info);
+                    storage.update(tasks);
+                    break;
+                case "UNMARK":
+                    info = Parser.parseInfo(input);
+                    handleUnmarking(info);
+                    storage.update(tasks);
+                    break;
+                case "DELETE":
+                    info = Parser.parseInfo(input);
+                    handleDelete(info);
+                    storage.update(tasks);
+                    break;
+                case "TODO":
+                    info = Parser.parseInfo(input);
+                    addTodo(info);
+                    storage.update(tasks);
+                    break;
+                case "EVENT":
+                    info = Parser.parseInfo(input);
+                    addEvent(info);
+                    storage.update(tasks);
+                    break;
+                case "DEADLINE":
+                    info = Parser.parseInfo(input);
+                    addDeadline(info);
+                    storage.update(tasks);
+                    break;
+                case "FIND":
+                    info = Parser.parseInfo(input);
+                    handleFind(info);
+                    break;
+                default:
+                    throw new DukeInvalidCommandException(command);
+                }
+            } catch (Exception e) {
+                ui.printError(e);
+            }
+        }
+    }
+
+    /**
+     * Stops the Duke chatbot.
+     */
+    public void handleExit() {
+        ui.printExit();
+        hasNext = false;
+    }
+
+    /**
+     * Generates list of added tasks.
+     */
+    public void handleList() {
```

Comment:
Not sure I like the naming of this method since it doesn't give me the impression that it is meant to generate the list